src/fragments/lib/graphqlapi/js/optimistic-ui.mdx

```diff
@@ -0,0 +1,1049 @@
+The GraphQL API category can be used with [TanStack Query](https://tanstack.com/query/latest/docs/react/overview) to implement optimistic UI, allowing CRUD operations to be rendered immediately on the UI before the request roundtrip has completed. Using the GraphQL API with TanStack additionally makes it easy to render loading and error states, and allows you to rollback changes on the UI when API calls are unsuccessful.
+
+In the following examples we'll create a list view that optimistically renders newly created items, and a detail view that optimistically renders updates and deletes.
+
+<Callout>
+
+For more details on TanStack Query, including requirements, supported browsers, and advanced usage, see the [TanStack Query documentation](https://tanstack.com/query/latest/docs/react/overview). 
+For complete guidance on how to implement optimistic updates with TanStack Query, see the [TanStack Query Optimistic UI Documentation](https://tanstack.com/query/latest/docs/react/guides/optimistic-updates).
+For more on the Amplify GraphQL API, see the [API documentation](https://docs.amplify.aws/lib/graphqlapi/getting-started/q/platform/js/).
+
+</Callout>
+
+To get started, run the following command in an existing Amplify project:
+
+```bash
+# Install TanStack Query
+npm i @tanstack/react-query
+
+# Select default configuration
+amplify add api              
+```
+
+When prompted, use the following schema, which can also be found under `amplify/backend/api/[name of project]/schema.graphql`:
+
+```graphql
+type RealEstateProperty @model @auth(rules: [{ allow: public }]) {
+  id: ID!
+  name: String!
+  address: String
+}
+```
+
+Save the schema and run `amplify push` to deploy the changes. For the purposes of this guide, we'll build a Real Estate Property listing application.
+
+Next, at the root of your project, add the required TanStack Query imports, and create a client:
+
+<Callout>
+
+TanStack Query Devtools are not required, but are a useful resource for debugging and understanding how TanStack works under the hood. By default, React Query Devtools are only included in bundles when `process.env.NODE_ENV === 'development'`, meaning that no additional configuration is required to exclude them from a production build. 
+For more information on the TanStack Query Devtools, visit the [TanStack Query Devtools docs](https://tanstack.com/query/latest/docs/react/devtools)
+
+</Callout>
```

Comment:
I'd move this above the "complete examples" call out. It's a bit too prominent up here. It's more useful once the code has been copy/pasted.

Reply:
I put this here so that the reader understands why we're importing the Devtools in the code snippet that follows. I don't feel strongly either way, so will move.